### ISSUE TITLE
continuum_mechanics: Beam.deflection() fails for indeterminate problem

### ISSUE THREAD

**Issue by ltroj** [sympy/sympy#22443]
When trying to solve the following system with an indeterminacy of 1 an error is issued when calculating the beams `deflection()`. However, all other quantities (`reaction_loads`, `shear_force`, `bending_moment`) can be calculated.

```python
from sympy.physics.continuum_mechanics.beam import Beam
from sympy import symbols
    
b = Beam(length=10000.,
         elastic_modulus=210000,
         second_moment=7.3808551e+07)

# Restrict 3 DOF
b.apply_support(loc=7000., type='roller')
b.apply_support(loc=0., type='pin')

# Add 4th boundary condition
R_NH = symbols('R_NH')
b.apply_load(R_NH, 5000., -1)
b.bc_deflection.append((5000., 0))

# Add point load
b.apply_load(32000., 4000., -1)

# The following works:
b.solve_for_reaction_loads(*(i[0] for i in b.applied_loads))
b.shear_force()
b.plot_shear_force()
b.bending_moment()
b.plot_bending_moment()

# Computation of deflection causes an error:
b.deflection()
```

### Traceback:
```python

>>> File "\sympy\physics\continuum_mechanics\beam.py", line 1220, in deflection
>>>    deflection_curve = deflection_curve.subs({C3: constants[0][0], C4: constants[0][1]})

>>> IndexError: list index out of range
```

**Comment by smichr** [comment]
The equations to be solved have no solution, but the code assumes that they do. Here are the equations:
```
[7000.0*C3 + C4 - 83200000000000.0, C4, 5000.0*C3 + C4 - 59428571428571.4]
```
Solving the first and last gives
```
{C3: 11885714285.7143, C4: -0.100000000000000}
```
But the 2nd equations requires that `C4=0` -- so the system has no solution.

**Comment by ltroj** [comment]
> But the 2nd equations requires that `C4=0` -- so the system has no solution.

Thanks for checking Christopher. 
For better understanding: this means there must be something wrong with the boundary conditions I defined?
I assumed that if the moment curve, shear force curve and reactions of a system are known, a bending line can also be calculated.

**Comment by oscarbenjamin** [comment]
> For better understanding: this means there must be something wrong with the boundary conditions I defined?

You have described the problem as *indeterminate* which would imply that there are infinitely many solutions but those equations suggest that the problem is *inconsistent* which means that there are no solutions.

**Comment by ltroj** [comment]
So maybe I messed up the terminology here. The problem is statically overdetermined.
I'm struggling because the problem is solvable in a number of other beam analysis tools I tried. Attached is a screenshot from [Beamax](https://github.com/lutzroeder/beamax). Shear force and moment curve are consistent with sympy's Beam results. Also the reactions sympy calculates are correct. It's only the deflection which cannot be calculated.
I am at a loss as to why the problem should be unsolvable in sympy's Beam and how to redefine the problem in sympy to get the correct eqns.

You sure this isn't a bug?

![github_issue_ex](https://user-images.githubusercontent.com/25671166/140910411-66359e17-cc30-49d5-a18f-6ce389a7e43d.png)

**Comment by oscarbenjamin** [comment]
> You sure this isn't a bug?

No, I'm not sure at all. I'm waiting for someone with better knowledge of this module to comment.

**Comment by moorepants** [comment]
I don't recall that SymPy Beam solves indeterminate problems automatically. You likely have to provide some manual steps to properly formulate and solve the problem, as explained in a solid mechanics textbook.

**Comment by skane88** [comment]
Could this be an issue to do with floating  point numbers? I have just come across a very similar issue with a 2 span continuous beam (3x pin supports, and 1x point load). It solves fine when I replace all floats with Rational() calls, but if I have the original floats it only determines the shear and moments and fails on the slope & deflection equations.

**Comment by skane88** [comment]
I have re-run the example from @ltroj using calls to ``Rational`` around all numeric input and it seems to solve fine, giving the solution:

```python
>>> 8320000*x/10849856997 - 136*SingularityFunction(x, 0, 3)/4068696373875 + 8*SingularityFunction(x, 4000, 3)/23249693565 - 232*SingularityFunction(x, 5000, 3)/581242339125 + 8*SingularityFunction(x, 7000, 3)/90415474975
```

**Comment by ltroj** [comment]
@skane88 Thanks for coming up with this. It seems to do the trick for me as well. I'd still consider that as unexpected behaviour from sympy(?)

**Comment by skane88** [comment]
@ltroj I don't know enough about sympy to know if this is expected or unexpected behavior. I was just having a look through the issues on GitHub and noticed that using ``Rational()`` was mentioned as a potential solution in issue #21162. It fixed my problem so I thought I'd see if it would fix your problem as well. Perhaps look at the discussions on #21162, #21377 for more background?
